### PR TITLE
[Perf][FP8_LIGHTING_INDEXER] drop logits clone and bound clean_logits scan

### PR DESCRIPTION
**Pull request by Ruihan11**
## Summary

Ref: #1689 

- Declare `mutates_args=("Logits",)` on the indexer custom op and return the caller-owned
  `Logits` instead of `Logits.clone()` — removes a 134 MB fp32 device-to-device round-trip per
  call (the clone existed only to satisfy the previous `mutates_args=()` declaration). Also
  fixes `register_fake` to match the new None-return schema (its previous fake shape was wrong).
- Rewrite `clean_logits_kernel` to iterate only the two out-of-window spans `[0, ks)` /
  `[ke, seq_len_kv)` instead of predicate-scanning all `seq_len × seq_len_kv` positions; adds
  the same `seq_len_kv` clamp on ks/ke that the main kernel already applies.
- Code-only: no numerical change, no Op signature / manifest / autotune-space change.

## Test plan

- [x] pre-commit passed (ruff, codespell, gitleaks)
- [x] `pytest tests/ops/test_fp8_lighting_indexer.py` — 2 passed (strict isfinite-mask compare
      on ragged varlen windows; covers tune=True)
- [x] `pytest benchmarks/tests` (contract) — 8 passed
- [x] `python scripts/validate_manifest.py` — all checks passed

## Benchmark

**Environment**: NVIDIA H200, CUDA 13.0, PyTorch 2.10.0, TileLang 0.1.11+cuda.git5aaef0fd

### FP8LightingIndexerOp

| Shape (b, s, h, d, s_kv, g) | This PR (ms) | main (ms) | Speedup | BW (TB/s) |
| --------------------------- | ------------ | --------- | ------- | --------- |
| (1, 4096, 32, 64, 8192, 1)  | 0.1048       | 0.1709    | 1.63×   | 1.37      |
| (1, 2048, 16, 64, 4096, 1)  | 0.0731       | 0.0882    | 1.21×   | 0.49      |

**Takeaways:** per-kernel breakdown (torch profiler, large shape) shows the win is removed
waste, not a changed kernel: the 64.8 µs clone Memcpy is eliminated entirely; `clean_logits`
drops 18.46 → 5.77 µs on full-range masks (on the ragged bench masks it is write-dominated —
119 MB of semantically required `-inf` fills — and stays ~41 µs); the main kernel is untouched
(96.6 → 93.8 µs, noise). Correctness is bit-preserved: same kernels, same write set, no
numerical path change.

**Command:** `python -m pytest benchmarks/ops/bench_fp8_lighting_indexer.py -v`

### CODE REVIEW

File: tileops/kernels/fp8_lighting_indexer.py [hunk 1]
```diff
@@ -149,29 +149,35 @@ def clean_logits_kernel(
...
-            cu_k_e = CuSeqLenKE[bx]
-
-            for n_i in T.Pipelined(T.ceildiv(seq_len_kv, block_K)):
-                for k_i in T.serial(block_K // 512):
-                    idx = n_i * block_K + k_i * 512 + tx
-                    if idx < cu_k_s or idx >= cu_k_e:
-                        for g in T.serial(kv_group):
-                            Logits[by, bx, idx, g] = -T.infinity(dtype)
+            cu_k_s = T.min(CuSeqLenKS[bx], seq_len_kv)
+            cu_k_e = T.min(CuSeqLenKE[bx], seq_len_kv)
```
[the remark this answers is not in the document]
Comment (Ruihan11) [reply tileops/kernels/fp8_lighting_indexer.py]: fixed

**Review by zhen8838 — APPROVED** [review]
